devdoc/doublylinkedlist_requirements.md

```diff
@@ -104,3 +127,46 @@ extern PDLIST_ENTRY DList_RemoveHeadList(PDLIST_ENTRY listHead);
 **SRS_DLIST_06_013: [** DList_RemoveHeadList shall return listHead if that's the only item in the list. **]**
 
 Note: The Flink & Blink of the returned PDLIST_ENTRY shall be undefined.
+
+### DList_Find
+```c
+MOCKABLE_FUNCTION(, PDLIST_ENTRY, DList_Find, PDLIST_ENTRY, listHead, DLIST_MATCH_FUNCTION, matchFunction, const void*, matchContext);
+```
+
+**SRS_DLIST_43_001: [** `DList_Find` shall call `matchFunction` on each entry in the list defined by `listHead` along with `matchContext`. **]**
+
+**SRS_DLIST_43_002: [** If the call to `matchFunction` for an entry returns `DLIST_MATCH_FUNCTION_MATCHING`, `DList_Find` shall return that entry. **]**
```

Comment:
>_DLIST_43_002: [** If the call to `matchFunction` for an entry returns `DLIST_MATCH_FUNCTION_MAT [](http://example.com/codeflow?start=5&length=96)

What if ERROR is returned? What is specced to happen then?
I think you need to have a return enum for this function and an out arg that actually returns the found element.

Reply:
added specs to handle ERROR.